### PR TITLE
Support for "days of week" for KEB maintenance window schedule

### PR DESCRIPTION
**Pull request by berkanercan**
**Description**

Changes proposed in this pull request:

- Logic to calculate the next available day for the maintenance window.
- Unit tests

### CODE REVIEW

File: components/kyma-environment-broker/internal/orchestration/manager/manager.go [hunk 1]
```diff
@@ -88,9 +89,11 @@ func (m *orchestrationManager) resolveOperations(o *internal.Orchestration) ([]o
 		for _, r := range runtimes {
 			windowBegin := time.Time{}
 			windowEnd := time.Time{}
+
 			if o.Parameters.Strategy.Schedule == orchestration.MaintenanceWindow {
-				windowBegin, windowEnd = m.resolveWindowTime(r.MaintenanceWindowBegin, r.MaintenanceWindowEnd)
+				windowBegin, windowEnd = m.resolveWindowTime(r.MaintenanceWindowBegin, r.MaintenanceWindowEnd, r.MaintenanceDays)
```
Comment (ebensom) [inline components/kyma-environment-broker/internal/orchestration/manager/manager.go:94]: Here the plan is that `r.MaintenanceDays` will be filled by this point by `m.resolver.Resolve()` according to a specified maintenance window policy?

File: components/kyma-environment-broker/internal/orchestration/manager/manager.go [hunk 2]
```diff
@@ -216,13 +219,49 @@ func (m *orchestrationManager) resolveWindowTime(beginTime, endTime time.Time) (
 
 	// if time window has already passed we wait until next day
 	if start.Before(n) && end.Before(n) {
-		start = start.AddDate(0, 0, 1)
-		end = end.AddDate(0, 0, 1)
+		day := firstAvailableDay(n.Day(), convertSliceOfDaysToMap(availableDays))
+		start = start.AddDate(0, 0, int(math.Abs(float64(day - n.Day()))))
```
Comment (ebensom) [inline components/kyma-environment-broker/internal/orchestration/manager/manager.go:223]: The day difference logic mentioned in my previous comment should be used also here

File: components/kyma-environment-broker/common/orchestration/strategies/parallel.go [hunk 3]
```diff
@@ -122,8 +123,11 @@ func (p *ParallelOrchestrationStrategy) processOperation(op orchestration.Runtim
 	case orchestration.MaintenanceWindow:
 		// if time window for this operation has finished, we requeue and reprocess on next time window
 		if !op.MaintenanceWindowEnd.IsZero() && op.MaintenanceWindowEnd.Before(time.Now()) {
-			op.MaintenanceWindowBegin = op.MaintenanceWindowBegin.Add(p.rescheduleDelay)
-			op.MaintenanceWindowEnd = op.MaintenanceWindowEnd.Add(p.rescheduleDelay)
+			day := firstAvailableDay(time.Now().Day(), convertSliceOfDaysToMap(op.MaintenanceDays))
+			dayDifference := int(math.Abs(float64(day - time.Now().Day())))
+			op.MaintenanceWindowBegin = op.MaintenanceWindowBegin.Add(p.rescheduleDelay * time.Duration(dayDifference))
```
[the remark this answers is not in the document]
Comment (berkanercan) [reply components/kyma-environment-broker/common/orchestration/strategies/parallel.go]: The code snippet plans the delay at least a day ahead, causing the pod to crash. I have changed it with the following. I still need confirmation if it is correct. 

```
if p.rescheduleDelay > 0 {
    op.MaintenanceWindowBegin = op.MaintenanceWindowBegin.Add(p.rescheduleDelay)
    op.MaintenanceWindowEnd = op.MaintenanceWindowEnd.Add(p.rescheduleDelay)
} else {
    op.MaintenanceWindowBegin = op.MaintenanceWindowBegin.AddDate(0, 0, diff)
    op.MaintenanceWindowEnd = op.MaintenanceWindowEnd.AddDate(0, 0, diff)
}
```

**Pull request closed without merging**